base/applications/mspaint/winproc.cpp

```diff
@@ -575,6 +576,100 @@ LRESULT CMainWindow::OnSysColorChange(UINT nMsg, WPARAM wParam, LPARAM lParam, B
     return 0;
 }
 
+typedef ULONG (WINAPI *FN_MAPISendMail)(LHANDLE, ULONG_PTR, lpMapiMessage, FLAGS, ULONG);
+typedef ULONG (WINAPI *FN_MAPISendMailW)(LHANDLE, ULONG_PTR, lpMapiMessageW, FLAGS, ULONG);
+
+ULONG CMainWindow::OpenMailerWithAttachmentW(LPCWSTR pszPathName)
+{
+    CStringW strFileTitle;
+    if (PathFileExistsW(pszPathName)) // File does exist?
+    {
+        strFileTitle = PathFindFileNameW(pszPathName);
+    }
+    else // Not existing
+    {
+        // Delete the temporary file if any
+        if (g_szTempFile[0])
+            ::DeleteFileW(g_szTempFile);
+
+        // Get the name of a temporary file
+        WCHAR szTempDir[MAX_PATH];
+        ::GetTempPathW(_countof(szTempDir), szTempDir);
+        if (!::GetTempFileNameW(szTempDir, L"afx", 0, g_szTempFile))
+            return MAPI_E_FAILURE;
+
+        // Save it to the temporary file
+        HBITMAP hbm = imageModel.CopyBitmap();
+        BOOL ret = ::SaveDIBToFile(hbm, g_szTempFile, FALSE);
+        ::DeleteObject(hbm);
+
+        if (!ret)
+        {
+            g_szTempFile[0] = UNICODE_NULL;
+            return MAPI_E_FAILURE; // Failure
+        }
+
+        // Use the temporary file 
+        pszPathName = g_szTempFile;
+        strFileTitle.LoadString(IDS_DEFAULTFILENAME);
+        strFileTitle += L".png";
```

Comment:
This extension must correspond to the format the image is saved by the SaveDIBToFile -- This is not evident whether this is the case, knowing that saving images with mspaint can be done with different formats.

Reply:
In this situation, the file is not saved yet, thus we  use a temp file. The best format for unknown image is png.

Reply:
Yes, but where in the code is that output format specified? Since currently you save the image into the temp file (that has the name afxWHATEVER.tmp) ?

Reply:
Can you actually send me an e-mail with attachment by this code?
My address is katayama.hirofumi.mz@gmail.com .

Reply:
Sorry.

```c
        // if the file type is null, get the file type from extension
        const GUID *FileType = &guidFileType;
        if (IsGuidEqual(guidFileType, GUID_NULL))
        {
            LPCWSTR pszExt = GetFileExtension(pszNameW);
            FileType = FileTypeFromExtension(pszExt);
        }
```
This code cannot send it.

Reply:
```c
        if (IsGuidEqual(guidFileType, GUID_NULL))
        {
            LPCWSTR pszExt = GetFileExtension(pszNameW);
            FileType = FileTypeFromExtension(pszExt);
            if (!FileType)
            {
                FileType = &Gdiplus::ImageFormatPNG;
            }
        }
```
Now is OK.